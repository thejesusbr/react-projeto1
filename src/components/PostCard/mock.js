export const postCardPropsMock = {
  title: 'TÃ­tulo',
  body: 'Corpo',
  id: 1,
  cover: 'img/img.png',
};
