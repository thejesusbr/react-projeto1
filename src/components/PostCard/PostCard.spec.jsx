import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    // @ts-ignore
    const { debug } = render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: 'Título' })).toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: /Título/i }));

    expect(screen.getByText('Corpo'));
  });

  it('should match snapshot', () => {
    // @ts-ignore
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
