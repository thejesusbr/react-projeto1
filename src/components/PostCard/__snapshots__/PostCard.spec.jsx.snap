// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Título"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      1
      : 
      Título
    </h2>
    <p>
      Corpo
    </p>
  </div>
</div>
`;
